qcodes_contrib_drivers/drivers/NationalInstruments/PXIe_2597.py

```diff
@@ -0,0 +1,58 @@
+from typing import Optional, Dict
+from qcodes.utils.validators import Enum
+from .Switch import NI_Switch
+
+
+class NI_PXIe_2597(NI_Switch):
+    r"""
+    QCoDeS driver for National Instruments RF switch PXIe-2597. The device
+    connects the common "com" port to any of the 6 other ports, labeled
+    "ch1"..."ch6" by default. Use the ``name_mapping `` argument to alias the
+    channel names.
+
+    Args:
+        name: Qcodes name for this instrument
+        resource: Network address or VISA alias for the instrument.
+        name_mapping: Optional mapping from default channel names to custom
+            aliases
+        reset_device: whether to reset the device on initialization
+    """
+    def __init__(self, name: str, resource: str,
+                 name_mapping: Optional[Dict[str, str]] = None,
+                 reset_device: bool = False, **kwargs):
+
+        if name_mapping is not None:
+            # don't mutate external dict
+            name_mapping = name_mapping.copy()
+            name_mapping["com"] = "com"
+
+        super().__init__(name, resource, name_mapping, reset_device, **kwargs)
+
+        self.channels.lock()
```

Comment:
why locking channels only here?

Reply:
I think the idea was that subclasses could customize the channel list if necessary. But you're right, at least currently there's no reason why it's not in the superclass.